src/magicbox/QueriesProcessor.ts

```diff
@@ -0,0 +1,91 @@
+import { defaults } from 'underscore';
+
+export interface IQueriesProcessorOptions {
+  timeout: number;
+}
+
+export enum QueryProcessResultStatus {
+  Finished = 'finished',
+  TimedOut = 'timedout',
+  Overriden = 'overriden'
+}
+
+export type QueryProcessResult<ItemType> =
```

Comment:
> 1. Just from this PR, I cannot tell if the generic is needed? If so, just a capital letter is enough (e.g. T)

I believe this generic is needed since this class will be used to accumulate both suggestions and search result previews. In this case, this generic is needed to infer the type of `items`.

> 2. Could we standardize the object shape to always contain both the `status` and `items` properties?

Definitely, I'll fix that.

> 3. For the type of `status`, I would use `QueryProcessResultStatus` or maybe renaming to `ProcessingStatus`, `UnprocessedResultStatus`?

Very good point. I'll try to avoid variable types such as this one and I'll use the enum names you suggested.

> 4. For the name, maybe `IQueryResult`, `IUnprocessedResult`? I find `QueryProcess` is a little vague.

That's a much better idea, I'll rename them.

> 5. I think an interface would look nicer in this case since we are specifying properties.

Do you mean one interface that always has the `items` property or do you mean two interfaces (`IQueryResult` and `IUnprocessedResult`)? For now, I'll use two interfaces.

Reply:
> 1. Just from this PR, I cannot tell if the generic is needed? If so, just a capital letter is enough (e.g. T)

I believe this generic is needed since this class will be used to accumulate both suggestions and search result previews. In this case, this generic is needed to infer the type of `items`.

> 2. Could we standardize the object shape to always contain both the `status` and `items` properties?

Definitely, I'll fix that.

> 3. For the type of `status`, I would use `QueryProcessResultStatus` or maybe renaming to `ProcessingStatus`, `UnprocessedResultStatus`?

Very good point. 

> 4. For the name, maybe `IQueryResult`, `IUnprocessedResult`? I find `QueryProcess` is a little vague.

That's a much better idea, I'll rename them.

> 5. I think an interface would look nicer in this case since we are specifying properties.

Good point.